Bit plane encoder

kk60503 opened the pull request:
Still working on binding

kk60503 committed:
Add files via upload

kk60503 committed:
Upload source files

Upload source program

kk60503 committed:
# This is a combination of 11 commits.
# This is the 1st commit message:

Add files via upload
# This is the commit message #2:

Delete olaf/bpe directory
# This is the commit message #3:

Add files via upload
# This is the commit message #4:

Delete olaf/bpe_wd directory
# This is the commit message #5:

Delete olaf/bpe/setup.py
# This is the commit message #6:

Delete olaf/bpe/test.py
# This is the commit message #7:

Delete olaf/bpe/CMakeLists.txt
# This is the commit message #8:

Add files via upload
# This is the commit message #9:

Add files via upload
# This is the commit message #10:

Add files via upload
# This is the commit message #11:

Add files via upload

ryanpdx commented on olaf/bpe/bpe.cpp:
```diff
@@ -0,0 +1,65 @@
+    //docstring
+    m.doc() = "pybind11 bpe plugin";    
+        
+        py::class_<BITSTREAM>(m, "Bitstream")
+            .def(py::init<unsigned long, unsigned long, unsigned int, 
+                                            unsigned int, unsigned char>());
```
This should be:
```
py::class_<BitStream>(m, "Bitstream")
        .def(py::init<>())
        .def_readonly("SegBitCounter", &BitStream::SegBitCounter)
        .def_readonly("TotalBitCounter", &BitStream::TotalBitCounter)
        .def_readonly("ByteBitCounter", &BitStream::ByteBuffer_4Bytes)
        .def_readonly("CodeWordAlighmentBits", &BitStream::CodeWordAlighmentBits)
        .def_readonly("CodeWord_Length", &BitStream::CodeWord_Length)
        .def_readonly("F_Bits", &BitStream::F_Bits);
```

ryanpdx commented on olaf/bpe/bpe.cpp:
```diff
@@ -0,0 +1,65 @@
+#include <Python.h>
+#include <structmember.h>
+#include <iostream>
+#include "pybind11/pybind11.h"
```
This is a global include, so use "<" and ">"

ryanpdx commented on olaf/bpe/bpe.cpp:
```diff
@@ -0,0 +1,65 @@
+#include <Python.h>
+#include <structmember.h>
+#include <iostream>
```
Not needed

ryanpdx commented on olaf/bpe/bpe.cpp:
```diff
@@ -0,0 +1,65 @@
+#include <structmember.h>
+#include <iostream>
+#include "pybind11/pybind11.h"
+#include <pybind11/pytypes.h>
+#include <pybind11/numpy.h>
+#include <string>
```
Not needed

ryanpdx commented on olaf/bpe/bpe.cpp:
```diff
@@ -0,0 +1,65 @@
+#include <pybind11/pytypes.h>
+#include <pybind11/numpy.h>
+#include <string>
+#include "source/main_pybind.h" 
```
Shouldn't this be `"source/globals.h"`?

ryanpdx commented on olaf/bpe/bpe.cpp:
```diff
@@ -0,0 +1,65 @@
+#include "source/main_pybind.h" 
+
+namespace py = pybind11;
+using namespace pybind11::literals;
```
Is this needed?

ryanpdx commented on olaf/bpe/makefile:
```diff
@@ -0,0 +1,79 @@
+#Makefile for binding bpe C code to Python
```
Should be Makefile not makefile.

ryanpdx commented on olaf/bpe/source/README.rtf:
```diff
@@ -0,0 +1,7 @@
+Kyle Klein, Ryan Medick for OreSat - August 2023
+kleinky@pdx.edu - oresat@pdx.edu
+
+This program has been adapted from the University of Nebraska-Lincoln's 
+implementation of the CCSDS 122.0-B-2 Recommended Standard image compression algorithm.
+
+
```
Remove my name and add a link to the website.

ryanpdx replied:
Use md not rtf

ryanpdx commented on olaf/bpe/source/makefile:
```diff
@@ -0,0 +1,21 @@
+# CCSDS image compression standard makefile for usage compiling as C code 
```
Makefile not makefile

ryanpdx commented on olaf/bpe/source/UNL_readme_kielymods.rtf:
```diff
@@ -0,0 +1,424 @@
+{\rtf1\mac\ansicpg10000\cocoartf824\cocoasubrtf480
```
Use the txt version, not rtf one

ryanpdx replied:
The pdf is not needed.

ryanpdx commented on olaf/bpe/source/pixel_check.py:
```diff
@@ -0,0 +1,30 @@
+# pixel_check.py: a simple program for comparing/checking each pixel in identical images
```
Remove this file

ryanpdx commented on olaf/bpe/source/main_pybind.h:
```diff
@@ -0,0 +1,330 @@
+// Header file including global.h plus binding declarations
```
Why does this exist, isn't it just `globals.h`?

ryanpdx commented on olaf/bpe/makefile:
```diff
@@ -0,0 +1,79 @@
+
+LINK_TARGET = bpe.so
+
+CPPFLAGS = \
```
Rename to `INC_DIRS`

ryanpdx commented on olaf/bpe/source/cfc.py:
```diff
@@ -0,0 +1,28 @@
+# A simple program to convert raw cfc images to viewable file types
```
Remove this file

ryanpdx commented on olaf/bpe/source/errorhandle.c:
```diff
@@ -0,0 +1,58 @@
+
+void DebugInfo(char *m)
+{
+#ifdef DEBUG
+	fprintf(stderr, m);
+#endif
```
This give compile errors

```
source/errorhandle.c: In function ‘DebugInfo’:
source/errorhandle.c:48:9: error: format not a string literal and no format arguments [-Werror=format-security]
   48 |         fprintf(stderr, m);
      |         ^~~~~~~
```

ryanpdx replied:
Commenting out #define DEBUG fixes it

ryanpdx commented on olaf/bpe/makefile:
```diff
@@ -0,0 +1,79 @@
+
+LDFLAGS = -lm $(shell python3-config --ldflags) \
+		  -undefined dynamic_lookup \
+		  #$(shell python3 -m pybind11 --includes)
```
Remove commented out line

ryanpdx commented on olaf/bpe/makefile:
```diff
@@ -0,0 +1,79 @@
+		 $(shell python3 -m pybind11 --includes)
+
+LDFLAGS = -lm $(shell python3-config --ldflags) \
+		  -undefined dynamic_lookup \
```
Code won't compile with the `-undefined dyanmic_lookup` flag on my system. If it is removed, it compiles just fine.

ryanpdx reviewed (commented):
Looks good, some clean up and changes to binding code needed.

kk60503 committed:
Add files via upload

Delete olaf/bpe directory

Add files via upload

Delete olaf/bpe_wd directory

Delete olaf/bpe/setup.py

Delete olaf/bpe/test.py

Delete olaf/bpe/CMakeLists.txt

Add files via upload

Add files via upload

Add files via upload

Add files via upload

Delete olaf/bpe/makefile

Delete olaf/bpe/makefile

Delete olaf/bpe/source/UNL_readme_kielymods.rtf

Add files via upload

it works

Add files via upload

Did some thorough combing of source files. The py bpe is now functional. Will begin testing/debug

Delete olaf/imgs_raw directory

Delete olaf/bpe directory

Add files via upload

reupload (wrong dir)

Delete olaf/source directory

Delete olaf/test directory

Delete olaf/Makefile

Delete olaf/README.md

Delete olaf/bpe.cpp

Delete olaf/bpe_test.py

Delete olaf/main_pybind.cpp

Delete olaf/bpe/source/test directory

Delete olaf/bpe/source/imgs_raw directory

Delete olaf/bpe/imgs_raw/RR.out

Delete olaf/bpe/imgs_raw/sensin.out

Update ycol_global.h

Delete olaf/bpe/test/imgs_conv/sensin.img.png

Update README.md

Update README.md

Update Makefile

kk60503 committed:
Add files via upload

Delete olaf/bpe directory

Add files via upload

Delete olaf/bpe_wd directory

Delete olaf/bpe/setup.py

Delete olaf/bpe/test.py

Delete olaf/bpe/CMakeLists.txt

Add files via upload

Add files via upload

Add files via upload

Add files via upload

Delete olaf/bpe/makefile

Delete olaf/bpe/makefile

Delete olaf/bpe/source/UNL_readme_kielymods.rtf

Add files via upload

it works

Add files via upload

Did some thorough combing of source files. The py bpe is now functional. Will begin testing/debug

Delete olaf/imgs_raw directory

Delete olaf/bpe directory

Add files via upload

reupload (wrong dir)

Delete olaf/source directory

Delete olaf/test directory

Delete olaf/Makefile

Delete olaf/README.md

Delete olaf/bpe.cpp

Delete olaf/bpe_test.py

Delete olaf/main_pybind.cpp

Delete olaf/bpe/source/test directory

Delete olaf/bpe/source/imgs_raw directory

Delete olaf/bpe/imgs_raw/RR.out

Delete olaf/bpe/imgs_raw/sensin.out

Update ycol_global.h

Delete olaf/bpe/test/imgs_conv/sensin.img.png

Update README.md

Update README.md

Update Makefile

kk60503 committed:
Add files via upload

Add completed program and unittest for review

ryanpdx commented on olaf/bpe/testing/test_bpe.py:
```diff
@@ -0,0 +1,88 @@
+            os.remove(self._png_infile)
+        if os.path.isfile(self._png_outfile):
+            os.remove(self._png_outfile)
+
```
Remove, main is not required for unit tests

kk60503 replied:
Sorry Ryan, did you want me to remove lines 86-down at this part?

ryanpdx commented on olaf/bpe/testing/test_bpe.py:
```diff
@@ -0,0 +1,88 @@
+    # min encode params = (name of file to be encoded, name of newly encoded file,
+    #                                   width of image, height of image)
+    # min decode params = (name of encoded file to decode, name for the outfile)
+
```
Use """ comment sytle for all docstrings

ryanpdx commented on olaf/bpe/testing/test_bpe.py:
```diff
@@ -0,0 +1,88 @@
+# testing suite for bpe python module
```
change to """ comment style

ryanpdx commented on olaf/bpe/main_pybind.cpp:
```diff
@@ -0,0 +1,165 @@
+#include "source/global.h"                                                      
+#include <iostream>
+
+
+// Original program information
+void Usage()                                                                    
+{                                                                               
+     fprintf(stderr, "/*****   Bit Plane Encoder Using Wavelet Transform    ***/\n");
+     fprintf(stderr, "bpe [-e]/[-d] [Input_file_name] [-o Output_file_name] "
+             "[-r BitsPerPixel]\n");
+     fprintf(stderr, "\nParameters: \n");                                       
+     fprintf(stderr, "[-e]: encoding filename; \n");                            
+     fprintf(stderr, "[-d]: decoding filename; \n");                            
+     fprintf(stderr, "[-o]: provide ouput file name. \n");                      
+     fprintf(stderr, "[-w]: the number of pixels of each row. \n");             
+     fprintf(stderr, "[-h]: the number of pixels of each column. \n");          
+     fprintf(stderr, "[-r]: bits per pixel for encoding. "
+             "(by default it is 0 and encoded to the last bitplane\n");
+     fprintf(stderr, "[-b]: the number of bits of each pixel. By default it is 8.\n");
+     fprintf(stderr, "[-f]: byte order of a pixel, if it consists of more than one bytes.\n" 
+             "0 means little endian, 1 means big endian. Default value is 0.\n");
+     fprintf(stderr, "[-t]: wavelet transform. 1 is integer 9-7 DWT "
+             "and 0 is floating 9-7 DWT. By default it is integer DWT\n");
+     fprintf(stderr, "[-s]: the number of blocks in each segment. By default it is 256.\n");
+     fprintf(stderr, "eg 1: bpe -e sensin.img -o codes -r 1.0 -w 256 -h 256 -s 256  \n");
+     fprintf(stderr, "eg 2: bpe -d codes -o ss.img \n");                        
+     fprintf(stderr, "**********  Author: Hongqiang Wang  *****************\n");
+     fprintf(stderr, "**********  Department of Electrical Engineering ****\n");
+     fprintf(stderr, "**********  University of Nebraska -Lincoln  ********\n");
+     fprintf(stderr, "**********  March 9, 2008   *************************\n");
+     fprintf(stderr, "/****************************************************\n");
+     fprintf(stderr, "/*****  Altered Sept 2023 for OreSat/PSAS  **********\n");
+     fprintf(stderr, "/****************************************************\n");
+    return;                                                                     
+}                                                                               
```
Remove

ryanpdx commented on olaf/bpe/testing/pixel_check.py:
```diff
@@ -0,0 +1,38 @@
+# pixel_check.py: 
+#  a simple program for comparing/checking each pixel in identical images
+# **currently has only been tested on PNG image type
+# adapted from:
+#  https://stackoverflow.com/questions/62285511/use-numpy-to-quickly-iterate-over-pixels
+# (program utilizes PIL)
```
Use block comment style

ryanpdx commented on olaf/bpe/testing/pixel_check.py:
```diff
@@ -0,0 +1,38 @@
+
+    print('scan complete')
+
+# if calling from command line: python pixel_check.py <file1> <file2>
+if __name__ == '__main__':
+    file1 = sys.argv[1]
+    file2 = sys.argv[2]
+    pixel_check(file1, file2)
```
Use argparser

ryanpdx reviewed (changes requested):
- Remove all `.pyc` files
- move `olaf/bpe/testing/` contents  to the `tests/bpe/`
- move all images out of the `olaf/` dir
- What are the `GHplaceholder` files

kk60503 committed:
Delete tests/bpe/img_files/raw/sinan.img

move/delete files

ryanpdx commented:
If you can, please try to squash some of your commits and update your comments to better reflect what changed in the commits. 

Also run `black` and `isort` on your `.py` files.

kk60503 committed:
Add files via upload

mostly commenting and linting

kk60503 committed:
Add files via upload

kk60503 committed:
Add files via upload

kk60503 committed:
Merge branch 'master' into bit_plane_encoder

kk60503 committed:
Delete olaf/bpe/testing directory

kk60503 committed:
Add files via upload

kk60503 committed:
Delete bin_check.py

kk60503 committed:
Delete cfc.py

kk60503 committed:
Delete test_bpe.py

kk60503 committed:
Delete pixel_check.py

kk60503 committed:
linting

kk60503 committed:
Add files via upload

kk60503 committed:
Add files via upload

kk60503 committed:
Add files via upload

kk60503 committed:
Add files via upload

kk60503 committed:
Update global.h

kk60503 committed:
Update requirements.txt

kk60503 committed:
Update pyproject.toml

kk60503 committed:
Add files via upload

kk60503 committed:
Update requirements.txt

kk60503 committed:
Add files via upload

kk60503 committed:
Add files via upload

kk60503 committed:
Create __init__.py

kk60503 committed:
Update test_bpe.py

kk60503 committed:
Add files via upload

kk60503 committed:
Update test_bpe.py

kk60503 committed:
Update test_bpe.py
